queue/src/fiskaltrust.Middleware.Localization.QueueDEFAULT/Dummy/Services/XXSSCDProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using fiskaltrust.Middleware.Abstractions;
+using fiskaltrust.Middleware.Contracts.Models;
+using fiskaltrust.storage.V0;
+using Newtonsoft.Json;
+
+namespace fiskaltrust.Middleware.Localization.QueueDEFAULT.Dummy.Services
+{
+    // The XXSSCDProvider class is responsible for managing the Signature Creation Units for the specific market "XX" (replace with the actual market name).
+    /// It provides functionality to register, retrieve, and manage the SCU instances for the given market, ensuring thread-safe access.
+    public class XXSSCDProvider : IXXSSCDProvider
+    {
+
+        private readonly IClientFactory<object> _clientFactory;
+        private readonly MiddlewareConfiguration _middlewareConfiguration;
+
+        private readonly SemaphoreSlim _semaphoreInstance = new SemaphoreSlim(1, 1);
+        private readonly SemaphoreSlim _semaphoreRegister = new SemaphoreSlim(1, 1);
+
+        private Object _instance;
+
+        public object Instance
+        {
+            get
+            {
+                try
+                {
+                    _semaphoreInstance.Wait();
+                    if (_instance == null)
+                    {
+                        RegisterCurrentScuAsync().Wait();
+                    }
+
+                    return _instance;
+                }
+                finally
+                {
+                    _semaphoreInstance.Release();
+                }
+            }
+        }
+
+        public XXSSCDProvider(IClientFactory<object> clientFactory, MiddlewareConfiguration middlewareConfiguration)
+        {
+            _clientFactory = clientFactory;
+            _middlewareConfiguration = middlewareConfiguration;
+        }
+
+        public async Task RegisterCurrentScuAsync()
+        {
+            try
+            {
+                await _semaphoreRegister.WaitAsync().ConfigureAwait(false);
+                var ftSignaturCreationUnitXX = JsonConvert.DeserializeObject<List<ftSignaturCreationUnitDE>>(_middlewareConfiguration.Configuration["init_ftSignaturCreationUnitXX"].ToString());
+
+                var uri = GetUriForSignaturCreationUnit(ftSignaturCreationUnitXX.FirstOrDefault().Url);
```

Comment:
```suggestion
                await _semaphoreRegister.WaitAsync().ConfigureAwait(false);
                var ftSignaturCreationUnitXX = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(_middlewareConfiguration.Configuration["init_ftSignaturCreationUnitXX"].ToString());

                var uri = GetUriForSignaturCreationUnit(ftSignaturCreationUnitXX.FirstOrDefault()["Url"]);
```

Thats just pseudo code not sure if it works exactly like that but like this way we could manage to not have the `ftSignaturCreationUnitDE` in here.